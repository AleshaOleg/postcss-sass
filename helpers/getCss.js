var fs = require('fs');
var postcss = require('postcss');

module.exports = function (fileName) {
    return postcss.parse(fs.readFileSync(
        './__tests__/css/' + fileName + '.css',
        'utf-8'
        )
    );
};
