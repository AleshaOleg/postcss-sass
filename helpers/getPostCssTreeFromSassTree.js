var fs = require('fs');
var sassToPostCss = require('../');

module.exports = function (fileName) {
    var source = fs.readFileSync(
        './__tests__/sass/' + fileName + '.sass',
        'utf-8'
    );
    return sassToPostCss(source);
};
