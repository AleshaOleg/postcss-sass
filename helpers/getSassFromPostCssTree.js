var fs = require('fs');
var postcss = require('postcss');
var postcssSass = require('../');

module.exports = function (fileName) {
    var source = fs.readFileSync(
        './__tests__/sass/' + fileName + '.sass',
        'utf-8'
    );

    return postcss().process(
        source,
        {
            syntax: postcssSass
        }).then(function (result) {
        return result.content; // Sass with transformations
    });
};

