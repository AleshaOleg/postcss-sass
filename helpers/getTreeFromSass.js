var fs = require('fs');
var gonzales = require('gonzales-pe');

module.exports = function (fileName) {
    var source = fs.readFileSync(
        './__tests__/sass/' + fileName + '.sass',
        'utf-8'
    );

    return gonzales.parse(source, {syntax: 'sass', rule: 'declaration'});
};
