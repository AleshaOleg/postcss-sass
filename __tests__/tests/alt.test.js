var getPostCssTreeFromSass =
    require('../../helpers/getPostCssTreeFromSassTree');

it('alt.sass', function () {
    expect(getPostCssTreeFromSass('alt')).toMatchSnapshot();
});
