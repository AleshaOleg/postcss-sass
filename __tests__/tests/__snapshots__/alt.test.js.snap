exports[`test alt.sass 1`] = `
Object {
  "nodes": Array [
    Object {
      "nodes": Array [
        Object {
          "prop": "float",
          "raws": Object {
            "before": "",
            "between": ": ",
            "semicolon": false,
          },
          "source": Object {
            "end": Object {
              "column": 14,
              "line": 2,
            },
            "input": Input {
              "css": "h1
  :float  left
  :width  274px
  height: 75px
  margin: 0
  background:
    repeat: no-repeat
    :image none
  a:hover, a:visited
    color: green
  b:hover
    color: red
    :background-color green
  const
    nosp: 1 + 2
    sp : 1 + 2
",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 3,
              "line": 2,
            },
          },
          "type": "decl",
          "value": "left",
        },
        Object {
          "prop": "width",
          "raws": Object {
            "before": "",
            "between": ": ",
            "semicolon": false,
          },
          "source": Object {
            "end": Object {
              "column": 15,
              "line": 3,
            },
            "input": Input {
              "css": "h1
  :float  left
  :width  274px
  height: 75px
  margin: 0
  background:
    repeat: no-repeat
    :image none
  a:hover, a:visited
    color: green
  b:hover
    color: red
    :background-color green
  const
    nosp: 1 + 2
    sp : 1 + 2
",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 3,
              "line": 3,
            },
          },
          "type": "decl",
          "value": "274px",
        },
        Object {
          "prop": "height",
          "raws": Object {
            "before": "",
            "between": ": ",
            "semicolon": false,
          },
          "source": Object {
            "end": Object {
              "column": 14,
              "line": 4,
            },
            "input": Input {
              "css": "h1
  :float  left
  :width  274px
  height: 75px
  margin: 0
  background:
    repeat: no-repeat
    :image none
  a:hover, a:visited
    color: green
  b:hover
    color: red
    :background-color green
  const
    nosp: 1 + 2
    sp : 1 + 2
",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 3,
              "line": 4,
            },
          },
          "type": "decl",
          "value": "75px",
        },
        Object {
          "prop": "margin",
          "raws": Object {
            "before": "",
            "between": ": ",
            "semicolon": false,
          },
          "source": Object {
            "end": Object {
              "column": 11,
              "line": 5,
            },
            "input": Input {
              "css": "h1
  :float  left
  :width  274px
  height: 75px
  margin: 0
  background:
    repeat: no-repeat
    :image none
  a:hover, a:visited
    color: green
  b:hover
    color: red
    :background-color green
  const
    nosp: 1 + 2
    sp : 1 + 2
",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 3,
              "line": 5,
            },
          },
          "type": "decl",
          "value": "0",
        },
        Object {
          "prop": "background-repeat",
          "raws": Object {
            "before": "",
            "between": ": ",
            "semicolon": false,
          },
          "source": Object {
            "end": Object {
              "column": 21,
              "line": 7,
            },
            "input": Input {
              "css": "h1
  :float  left
  :width  274px
  height: 75px
  margin: 0
  background:
    repeat: no-repeat
    :image none
  a:hover, a:visited
    color: green
  b:hover
    color: red
    :background-color green
  const
    nosp: 1 + 2
    sp : 1 + 2
",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 5,
              "line": 7,
            },
          },
          "type": "decl",
          "value": "no-repeat",
        },
        Object {
          "prop": "background-image",
          "raws": Object {
            "before": "",
            "between": ": ",
            "semicolon": false,
          },
          "source": Object {
            "end": Object {
              "column": 15,
              "line": 8,
            },
            "input": Input {
              "css": "h1
  :float  left
  :width  274px
  height: 75px
  margin: 0
  background:
    repeat: no-repeat
    :image none
  a:hover, a:visited
    color: green
  b:hover
    color: red
    :background-color green
  const
    nosp: 1 + 2
    sp : 1 + 2
",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 5,
              "line": 8,
            },
          },
          "type": "decl",
          "value": "none",
        },
      ],
      "raws": Object {
        "between": "
",
      },
      "selector": "h1",
      "source": Object {
        "end": Object {
          "column": 15,
          "line": 16,
        },
        "input": Input {
          "css": "h1
  :float  left
  :width  274px
  height: 75px
  margin: 0
  background:
    repeat: no-repeat
    :image none
  a:hover, a:visited
    color: green
  b:hover
    color: red
    :background-color green
  const
    nosp: 1 + 2
    sp : 1 + 2
",
          "id": "<input css 1>",
        },
        "start": Object {
          "column": 1,
          "line": 1,
        },
      },
      "type": "rule",
    },
    Object {
      "nodes": Array [
        Object {
          "prop": "color",
          "raws": Object {
            "before": "",
            "between": ": ",
            "semicolon": false,
          },
          "source": Object {
            "end": Object {
              "column": 16,
              "line": 10,
            },
            "input": Input {
              "css": "h1
  :float  left
  :width  274px
  height: 75px
  margin: 0
  background:
    repeat: no-repeat
    :image none
  a:hover, a:visited
    color: green
  b:hover
    color: red
    :background-color green
  const
    nosp: 1 + 2
    sp : 1 + 2
",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 5,
              "line": 10,
            },
          },
          "type": "decl",
          "value": "green",
        },
      ],
      "raws": Object {
        "between": " 
",
      },
      "selector": "h1 a:hover, a:visited",
      "source": Object {
        "end": Object {
          "column": 17,
          "line": 10,
        },
        "input": Input {
          "css": "h1
  :float  left
  :width  274px
  height: 75px
  margin: 0
  background:
    repeat: no-repeat
    :image none
  a:hover, a:visited
    color: green
  b:hover
    color: red
    :background-color green
  const
    nosp: 1 + 2
    sp : 1 + 2
",
          "id": "<input css 1>",
        },
        "start": Object {
          "column": 3,
          "line": 9,
        },
      },
      "type": "rule",
    },
    Object {
      "nodes": Array [
        Object {
          "prop": "color",
          "raws": Object {
            "before": "",
            "between": ": ",
            "semicolon": false,
          },
          "source": Object {
            "end": Object {
              "column": 14,
              "line": 12,
            },
            "input": Input {
              "css": "h1
  :float  left
  :width  274px
  height: 75px
  margin: 0
  background:
    repeat: no-repeat
    :image none
  a:hover, a:visited
    color: green
  b:hover
    color: red
    :background-color green
  const
    nosp: 1 + 2
    sp : 1 + 2
",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 5,
              "line": 12,
            },
          },
          "type": "decl",
          "value": "red",
        },
        Object {
          "prop": "background-color",
          "raws": Object {
            "before": "",
            "between": ": ",
            "semicolon": false,
          },
          "source": Object {
            "end": Object {
              "column": 27,
              "line": 13,
            },
            "input": Input {
              "css": "h1
  :float  left
  :width  274px
  height: 75px
  margin: 0
  background:
    repeat: no-repeat
    :image none
  a:hover, a:visited
    color: green
  b:hover
    color: red
    :background-color green
  const
    nosp: 1 + 2
    sp : 1 + 2
",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 5,
              "line": 13,
            },
          },
          "type": "decl",
          "value": "green",
        },
      ],
      "raws": Object {
        "between": "
",
      },
      "selector": "h1 b:hover",
      "source": Object {
        "end": Object {
          "column": 28,
          "line": 13,
        },
        "input": Input {
          "css": "h1
  :float  left
  :width  274px
  height: 75px
  margin: 0
  background:
    repeat: no-repeat
    :image none
  a:hover, a:visited
    color: green
  b:hover
    color: red
    :background-color green
  const
    nosp: 1 + 2
    sp : 1 + 2
",
          "id": "<input css 1>",
        },
        "start": Object {
          "column": 3,
          "line": 11,
        },
      },
      "type": "rule",
    },
    Object {
      "nodes": Array [
        Object {
          "prop": "nosp",
          "raws": Object {
            "before": "",
            "between": ": ",
            "semicolon": false,
          },
          "source": Object {
            "end": Object {
              "column": 15,
              "line": 15,
            },
            "input": Input {
              "css": "h1
  :float  left
  :width  274px
  height: 75px
  margin: 0
  background:
    repeat: no-repeat
    :image none
  a:hover, a:visited
    color: green
  b:hover
    color: red
    :background-color green
  const
    nosp: 1 + 2
    sp : 1 + 2
",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 5,
              "line": 15,
            },
          },
          "type": "decl",
          "value": "3",
        },
        Object {
          "prop": "sp",
          "raws": Object {
            "before": "",
            "between": ": ",
            "semicolon": false,
          },
          "source": Object {
            "end": Object {
              "column": 14,
              "line": 16,
            },
            "input": Input {
              "css": "h1
  :float  left
  :width  274px
  height: 75px
  margin: 0
  background:
    repeat: no-repeat
    :image none
  a:hover, a:visited
    color: green
  b:hover
    color: red
    :background-color green
  const
    nosp: 1 + 2
    sp : 1 + 2
",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 5,
              "line": 16,
            },
          },
          "type": "decl",
          "value": "3",
        },
      ],
      "raws": Object {
        "between": "
",
      },
      "selector": "h1 const",
      "source": Object {
        "end": Object {
          "column": 15,
          "line": 16,
        },
        "input": Input {
          "css": "h1
  :float  left
  :width  274px
  height: 75px
  margin: 0
  background:
    repeat: no-repeat
    :image none
  a:hover, a:visited
    color: green
  b:hover
    color: red
    :background-color green
  const
    nosp: 1 + 2
    sp : 1 + 2
",
          "id": "<input css 1>",
        },
        "start": Object {
          "column": 3,
          "line": 14,
        },
      },
      "type": "rule",
    },
  ],
  "raws": Object {
    "after": "",
    "semicolon": false,
  },
  "source": Object {
    "end": Object {
      "column": 15,
      "line": 16,
    },
    "input": Input {
      "css": "h1
  :float  left
  :width  274px
  height: 75px
  margin: 0
  background:
    repeat: no-repeat
    :image none
  a:hover, a:visited
    color: green
  b:hover
    color: red
    :background-color green
  const
    nosp: 1 + 2
    sp : 1 + 2
",
      "id": "<input css 1>",
    },
    "start": Object {
      "column": 1,
      "line": 1,
    },
  },
  "type": "root",
}
`;
